### PR TITLE
Implement a Make action for updating schema.json 

### PR DESCRIPTION
**Pull request by frankban**
Also include a new schema generated with `make update-schema`. Please do not review schema.json, and it will become a pretty quick PR.

### CODE REVIEW

File: schemas/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Updating schema
```
Comment (hatched) [inline schemas/README.md]: I'm not sure this file is necessary, instead just populate the main README

File: generator/_generate.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def run(ns):
             name, version = facade['Name'], facade['Version']
         except (KeyError, TypeError, ValueError) as err:
             raise AppError('cannot retrieve name and version for facade')
-        if name in _BLACKLIST:
+        if name in _BLACKLIST or facade['Methods'] is None:
```
Comment (hatched) [inline generator/_generate.py]: curious why this is the case...can you add a code comment plz

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,8 @@
 DEVENV = devenv
 NODE_MODULES = node_modules
 PROJECT = js-libjuju
-SYSDEPS = build-essential python3-virtualenv tox
+SCHEMA = schemas/schema.json
```
Comment (hatched) [inline Makefile]: Can we have this save the schema with a version number of some sort? Even if it's the juju 6char sha?

File: schemas/README.md (hunk 1 above)
Comment (frankban) [reply schemas/README.md]: Done.

File: generator/_generate.py (hunk 2 above)
Comment (frankban) [reply generator/_generate.py]: Done.

File: Makefile (hunk 3 above)
Comment (frankban) [reply Makefile]: Yes, but it would complicate things a little bit. Maybe I could save a separate file with metadata about the schema I guess.

Comment (frankban) [reply Makefile]: Yes ok I'll do that in a follow-up.

**Pull request merged**